protocol/x/subaccounts/keeper/transfer.go

```diff
@@ -219,70 +218,61 @@ func (k Keeper) WithdrawFundsFromSubaccountToAccount(
 func (k Keeper) DistributeFees(
 	ctx sdk.Context,
 	assetId uint32,
-	quantums *big.Int,
-	perpetualId uint32,
+	revShares []revsharetypes.RevShare,
+	fill clobtypes.FillForProcess,
 ) error {
 	// get perpetual
-	perpetual, err := k.perpetualsKeeper.GetPerpetual(ctx, perpetualId)
+	totalFeeQuoteQuantums := new(big.Int).Add(fill.TakerFeeQuoteQuantums, fill.MakerFeeQuoteQuantums)
+	perpetual, err := k.perpetualsKeeper.GetPerpetual(ctx, fill.ProductId)
 	if err != nil {
 		return err
 	}
 
-	collateralPoolAddr, err := k.GetCollateralPoolFromPerpetualId(ctx, perpetualId)
+	collateralPoolAddr, err := k.GetCollateralPoolFromPerpetualId(ctx, fill.ProductId)
 	if err != nil {
 		return err
 	}
 
 	// calculate market mapper rev share
-	marketMapperShare := big.NewInt(0)
-	revShareAddr, revSharePpm, err := k.revShareKeeper.GetMarketMapperRevenueShareForMarket(
-		ctx,
-		perpetual.Params.MarketId,
-	)
-	// Note: The likelihood of this error is very low, and not getting the rev share should not
-	// prevent the trade from going through. Therefore, we log the error and continue
-	if err != nil {
-		log.ErrorLog(ctx, "DistributeFees: failed to get market mapper revenue share", "err", err)
-	}
-	if err == nil && revShareAddr != nil {
-		if revSharePpm >= 1e6 {
-			log.ErrorLog(
-				ctx,
-				"DistributeFees: revSharePpm is greater than or equal to 100%",
-				"revSharePpm",
-				revSharePpm,
-			)
-		} else {
-			// marketMapperShare = quantums * revSharePpm / 1e6
-			marketMapperShare.Div(
-				new(big.Int).Mul(quantums, big.NewInt(int64(revSharePpm))),
-				big.NewInt(1e6),
-			)
-		}
+	totalRevShareQuoteQuantums := big.NewInt(0)
+
+	for _, revShare := range revShares {
+		totalRevShareQuoteQuantums.Add(totalRevShareQuoteQuantums, revShare.QuoteQuantums)
 	}
 
 	// Remaining amount goes to the fee collector
-	feeCollectorShare := new(big.Int).Sub(quantums, marketMapperShare)
+	feeCollectorShare := new(big.Int).Sub(totalFeeQuoteQuantums, totalRevShareQuoteQuantums)
```

Comment:
Nit: explicitly check if `feeCollectorsShare` <= 0. If so panic

Reply:
Too many unrelated tests rely on feeCollectorsShare = 0. Setting it to < 0 rather than <=0

Reply:
> Too many unrelated tests rely on feeCollectorsShare = 0. 

Why is this the case? Wouldn't ` feeCollectorsShare = 0` indicate `totalFeeQuoteQuantums = totalRevShareQuoteQuantums`? Is it because in a lot of test cases both are `zero`? (We the protocol constraint that every fill results in protocol revenue)

Reply:
[Here](https://github.com/dydxprotocol/v4-chain/blob/affan/revshare-logic/protocol/x/clob/keeper/liquidations_test.go#L165) , [here](https://github.com/dydxprotocol/v4-chain/blob/affan/revshare-logic/protocol/x/clob/e2e/app_test.go#L431) and [here](https://github.com/dydxprotocol/v4-chain/blob/affan/revshare-logic/protocol/x/clob/abci_test.go#L1056) are some examples where that check fails.